javaHapiValidatorLambda/src/main/java/software/amazon/fwoa/Validator.java

```diff
@@ -0,0 +1,161 @@
+
+        // This module implements terminology services for in-memory code validation
+        supportChain.addValidationSupport(new InMemoryTerminologyServerValidationSupport(ctx));
+
+        // Create a PrePopulatedValidationSupport which can be used to load custom definitions.
+        PrePopulatedValidationSupport prepopulatedValidationSupport = loadIgs(ctx);
+        supportChain.addValidationSupport(prepopulatedValidationSupport);
+
+        // Create a validator using the FhirInstanceValidator module.
+        FhirInstanceValidator validatorModule = new FhirInstanceValidator(supportChain);
+        validator = ctx.newValidator().registerValidatorModule(validatorModule);
+    }
+
+    public ValidatorResponse validate(String resourceAsJsonText) {
+        try {
+            ValidationResult result = validator.validateWithResult(resourceAsJsonText);
+            return toValidatorResponse(result);
+        } catch (JsonSyntaxException | NullPointerException | IllegalArgumentException | InvalidRequestException e) {
+            return ValidatorResponse.builder()
+                .isSuccessful(false)
+                .errorMessages(ImmutableList.of(ValidatorErrorMessage.builder()
+                    .msg("Invalid JSON")
+                    .severity("error")
+                    .build()))
+                .build();
+        }
+    }
+
+    private ValidatorResponse toValidatorResponse(ValidationResult result) {
+        return ValidatorResponse.builder()
+            .isSuccessful(result.isSuccessful())
+            .errorMessages(result.getMessages().stream()
+                .map(singleValidationMessage -> ValidatorErrorMessage.builder()
+                    .severity(singleValidationMessage.getSeverity().getCode())
+                    .msg(singleValidationMessage.getLocationString() + " - " + singleValidationMessage.getMessage())
+                    .build())
+                .collect(Collectors.toList())
+            )
+            .build();
+    }
+
+    private PrePopulatedValidationSupport loadIgs(final FhirContext ctx) {
+
+        final ImmutableSet<String> allowedResourceTypes = ImmutableSet.of("StructureDefinition", "CodeSystem", "ValueSet");
+
+        IParser parser = ctx.newJsonParser();
+        parser.setParserErrorHandler(new StrictErrorHandler());
+        final PrePopulatedValidationSupport prePopulatedValidationSupport = new PrePopulatedValidationSupport(ctx);
+
+        try (ScanResult allFiles = new ClassGraph().acceptPaths(IMPLEMENTATION_GUIDES_FOLDER).rejectPaths(IMPLEMENTATION_GUIDES_FOLDER + "/*/*").scan()) {
+            ResourceList jsonResources = allFiles.getResourcesWithExtension("json");
+
+            ResourceList indexFiles = jsonResources.filter(x -> x.getPath().endsWith(".index.json"));
+
+            for (Resource indexFile : indexFiles) {
+                IgIndex igIndex = GSON.fromJson(indexFile.getContentAsString(), IgIndex.class);
+                for (IgFile file : igIndex.files) {
```

Comment:
Value in parallelization of these? How many files would we loop through here. The processing looks independent to each other